lib/gui/components/update-notifier.js

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+/**
+ * @module Etcher.Components.UpdateNotifier
+ */
+
+const angular = require('angular');
+const electron = require('electron');
+const Bluebird = require('bluebird');
+const _ = require('lodash');
+const settings = require('../models/settings');
+const units = require('../../shared/units');
+const release = require('../../shared/release');
+const packageJSON = require('../../../package.json');
+
+const MODULE_NAME = 'Etcher.Components.UpdateNotifier';
+const UpdateNotifier = angular.module(MODULE_NAME, [
+  require('../os/open-external/open-external'),
+  require('../modules/analytics')
+]);
+
+UpdateNotifier.service('UpdateNotifierService', function(AnalyticsService, OSOpenExternalService) {
+
+  /**
+   * @summary The number of days the update notifier can be put to sleep
+   * @constant
+   * @private
+   * @type {Number}
+   */
+  this.UPDATE_NOTIFIER_SLEEP_DAYS = packageJSON.updates.sleepDays;
+
+  /**
+   * @summary The current Electron browser window
+   * @constant
+   * @private
+   * @type {Object}
+   */
+  const currentWindow = electron.remote.getCurrentWindow();
+
+  /**
+   * @summary Determine if it's time to check for updates
+   * @function
+   * @public
+   *
+   * @param {Object} options - options
+   * @param {Number} [options.lastSleptUpdateNotifier] - last slept update notifier time
+   * @param {String} [options.lastSleptUpdateNotifierVersion] - last slept update notifier version
+   * @param {String} options.releaseType - release type
+   * @returns {Boolean} should check for updates
+   *
+   * @example
+   * if (UpdateNotifierService.shouldCheckForUpdates({
+   *   lastSleptUpdateNotifier: Date.now(),
+   *   lastSleptUpdateNotifierVersion: '1.0.0',
+   *   releaseType: release.RELEASE_TYPE.PRODUCTION
+   * })) {
+   *   console.log('We should check for updates!');
+   * }
+   */
+  this.shouldCheckForUpdates = (options) => {
+    if (_.some([
+      !options.lastSleptUpdateNotifier,
```

Comment:
The `options` object is not optional anymore, so its safe to use the dot notation. I usually only use `_.get` if we don't know whether the base object exists or not (`_.get` takes care of that automatically).